thor/orbits/od.py

```diff
@@ -133,23 +144,16 @@ def od(
     max_iter: int = 20,
     method: Literal["central", "finite"] = "central",
     fit_epoch: bool = False,
-    propagator: Literal["PYOORB"] = "PYOORB",
-    propagator_kwargs: dict = {},
+    propagator: Propagator = PYOORB(),
```

Comment:
Not a good idea to instantiate instances here. Instance will get instantiated when the module is imported and will always use the same instance in the same process, making it unpredictable.

I think it's okay to change the type definitions to accept a class, although I think a better approach here is to register each valid propagator class with a name key and do a dictionary lookup from a global adam_core.propagators module and instantiate inside the method.